Sources/AST/Environment.swift

```diff
@@ -350,6 +351,30 @@ public struct Environment {
       if binaryExpression.opToken.isBooleanOperator {
         return .basicType(.bool)
       }
+      if binaryExpression.opToken == .dot {
+        switch type(of: binaryExpression.lhs, enclosingType: enclosingType, callerCapabilities: callerCapabilities, scopeContext: scopeContext) {
+        case .arrayType(_):
+          if case .identifier(let identifier) = binaryExpression.rhs, identifier.name == "size" {
+            return .basicType(.int)
+          } else {
+            fatalError()
+          }
+        case .fixedSizeArrayType(_):
+          if case .identifier(let identifier) = binaryExpression.rhs, identifier.name == "size" {
+            return .basicType(.int)
+          } else {
+            fatalError()
+          }
+        case .dictionaryType(_):
+          if case .identifier(let identifier) = binaryExpression.rhs, identifier.name == "size" {
```

Comment:
Same as the prior discussion on the size. This needs to be generalised in future, but is sufficient for now